plaso/parsers/mactime.py

```diff
@@ -50,176 +56,257 @@ def __init__(self):
+      float: floating-point value or None if not available or invalid.
 
-      else:
-        date_time = dfdatetime_posix_time.PosixTime(timestamp=posix_time)
+    Raises:
+      UnableToParseFile: when an invalid floating-point value is found on
+          the first line.
+    """
+    float_value = values.pop(-1) or None
+    if float_value is not None:
+      try:
+        float_value = float(float_value)
+      except ValueError:
+        error_string = 'invalid {0:s} value in line: {1:d}'.format(
+            description, line_number)
+        if line_number == 0:
+          raise errors.UnableToParseFile(error_string)
+
+        parser_mediator.ProduceRecoveryWarning(error_string)
+        float_value = None
 
-      event = time_events.DateTimeValuesEvent(date_time, timestamp_description)
-      parser_mediator.ProduceEventWithEventData(event, event_data)
+    return float_value
 
-  # pylint: disable=unused-argument
-  def VerifyRow(self, parser_mediator, row):
-    """Verifies if a line of the file is in the expected format.
+  def ParseFileObject(self, parser_mediator, file_object):
+    """Parses a mactime bodyfile file-like object.
 
     Args:
-      parser_mediator (ParserMediator): mediates interactions between
-          parsers and other components, such as storage and dfvfs.
-      row (dict[str, str]): fields of a single row, as specified in COLUMNS.
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      file_object (dfvfs.FileIO): file-like object.
 
-    Returns:
-      bool: True if this is the correct parser, False otherwise.
+    Raises:
+      UnableToParseFile: when the file cannot be parsed.
     """
-    # Sleuthkit version 3 format:
-    # MD5|name|inode|mode_as_string|UID|GID|size|atime|mtime|ctime|crtime
-    # 0|/lost+found|11|d/drwx------|0|0|12288|1337961350|1337961350|1337961350|0
-
-    if row['md5'] != '0' and not self._MD5_RE.match(row['md5']):
-      return False
+    # Note that we cannot use the DSVParser here since the mactime bodyfile
+    # format is not strict and clean file format.
+    line_reader = text_file.TextFile(
+        file_object, encoding='UTF-8', end_of_line='\n')
 
-    # Check if the following columns contain a base 10 integer value if set.
-    for column_name in ('uid', 'gid', 'size'):
-      column_value = row.get(column_name, None)
-      if not column_value:
-        continue
+    line_number = 0
 
-      try:
-        int(column_value, 10)
-      except (TypeError, ValueError):
-        return False
+    try:
+      line = line_reader.readline()
+    except UnicodeDecodeError as exception:
+      raise errors.UnableToParseFile(
+          'unable to read line: {0:d} with error: {1!s}'.format(
+              line_number, exception))
+
+    while line:
+      values = line.split('|')
+      if len(values) < 11:
+        error_string = 'invalid number of values in line: {0:d}'.format(
```

Comment:
Please add the number of values actually found in the line to this error message

Reply:
done